### ISSUE TITLE
Inconsistent preprocess_params

### ISSUE THREAD

**Issue by HughSt** [disarm-platform/docs#9]
Hi all - so in the python template the `index.py` script is set up to use `preprocess_params.py` as a checker function only, i.e. it doesn't return anything. Accordingly, in `index.py` the line is run like this:

```
# Mutate the params to get them ready for use
        preprocess_params.preprocess(params)
```

However, in the R version, `main.r` catches the output from `preprocess_params.r`, e.g.
```
params <- preprocess_params(params)
```

That said, currently the `preprocess_params.r` doesn't prompt the user to return anything. Therefore, at this step in `main.r` params gets replaced with `NULL`. 

I think it is better to insist that `preprocess_params` *does* return something. This opens up the opportunity to actually mutate params if need be. If others agree, I think these are the things we need to do

- Change `preprocess_params` in both r and python templates to ensure they return something.
- Change `index.py` to catch the outgoing params
- Update the handholding docs to remind people to ensure that `preprocess_params` returns something.

Thoughts?

**Comment by onlyjsmith** [comment]
I'd prefer to _not_ mutate the params in the pre-process step - makes it easier to reason about.

We came across a similar issue with the Python template yesterday. You're right that the two templates are inconsistent, and we probably want them more similar. 

Have opened some related issues in the `faas-templates` repo:

- https://github.com/disarm-platform/faas-templates/issues/8
- https://github.com/disarm-platform/faas-templates/issues/7

**Comment by HughSt** [comment]
Yes I think we had it mutate for situations where you need to load something into memory in order to check it. If we don't mutate, then the function will need to load into memory once to check it and then again in the main function to use it. Are we OK to add this overhead, or should we mutate in certain situations with the default being not to mutate?
